spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/StreamDefinitionToDslConverter.java

```diff
@@ -112,9 +112,16 @@ public String toDsl(List<StreamAppDefinition> appDefinitions) {
 	}
 
 	private String autoQuotes(String propertyValue) {
-		if (!propertyValue.contains("'")) {
-			if (propertyValue.contains(" ") || propertyValue.contains(";")  || propertyValue.contains("*")) {
-				return "'" + propertyValue + "'";
+		if (!propertyValue.startsWith("\"") || !propertyValue.endsWith("\"")) {
```

Comment:
I think we only need to escape the characters that yaml doesn't escape already, otherwise we get an extra slash, e.g. \\t

Reply:
See escape chars in https://bitbucket.org/asomov/snakeyaml/src/c2ad14de14e7308ff8f26aa969e33c99dc5a44b8/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java?at=default&fileviewer=file-view-default#Emitter.java-1220